modyn/tests/supervisor/internal/triggers/performance/test_ensembletrigger.py

```diff
@@ -0,0 +1,270 @@
+    assert next_subtrigger_index_cache == {
+        "time_trigger": 21,
+        "data_amount_trigger": None,
+    }
+
+    # exhaust second
+    trigger._update_outdated_next_trigger_indexes(21, subtrigger_generators, next_subtrigger_index_cache)
+    assert next_subtrigger_index_cache == {
+        "time_trigger": None,
+        "data_amount_trigger": None,
+    }
+
+
+def test_find_next_trigger_index(
+    ensemble_trigger_config: EnsembleTriggerConfig,
+) -> None:
+    trigger = EnsembleTrigger(ensemble_trigger_config)
+
+    subtrigger_decision_cache = {"t1": False, "t2": False, "t3": False, "t4": False}
+    next_subtrigger_index_cache = {"t1": None, "t2": 10, "t3": 5, "t4": 3}
+
+    trigger.config.ensemble_strategy = AtLeastNEnsembleStrategy(n=2)
+
+    new_data = [(0, 0, 0)]  # dummy
+
+    trigger_log = TriggerPolicyEvaluationLog()
+
+    # trigger after second subtrigger
+    result = trigger._find_next_trigger_index(
+        processing_head=0,
+        new_data=new_data,
+        subtrigger_decision_cache=subtrigger_decision_cache,
+        next_subtrigger_index_cache=next_subtrigger_index_cache,
+        log=trigger_log,
+    )
+
+    assert result == 5
+    assert len(trigger_log.evaluations) == 2
+    assert [e.trigger_index for e in trigger_log.evaluations] == [3, 5]
+    assert [e.triggered for e in trigger_log.evaluations] == [False, True]
+
+    trigger_log = TriggerPolicyEvaluationLog()
+
+    # no trigger case, not 3 subtriggers
+    subtrigger_decision_cache = {"t1": False, "t2": False, "t3": False, "t4": False}
+    next_subtrigger_index_cache = {"t1": None, "t2": 10, "t3": 5, "t4": None}
+    trigger.config.ensemble_strategy = AtLeastNEnsembleStrategy(n=3)
+    result = trigger._find_next_trigger_index(
+        processing_head=0,
+        new_data=new_data,
+        subtrigger_decision_cache=subtrigger_decision_cache,
+        next_subtrigger_index_cache=next_subtrigger_index_cache,
+        log=trigger_log,
+    )
+    assert result is None
+    assert len(trigger_log.evaluations) == 2
+    assert [e.trigger_index for e in trigger_log.evaluations] == [5, 10]
+    assert [e.triggered for e in trigger_log.evaluations] == [False, False]
+
+    trigger_log = TriggerPolicyEvaluationLog()
+
+    # no subtrigger
+    subtrigger_decision_cache = {"t1": False, "t2": False, "t3": False, "t4": False}
+    next_subtrigger_index_cache = {"t1": None, "t2": None, "t3": None, "t4": None}
+    result = trigger._find_next_trigger_index(
+        processing_head=0,
+        new_data=new_data,
+        subtrigger_decision_cache=subtrigger_decision_cache,
+        next_subtrigger_index_cache=next_subtrigger_index_cache,
+        log=trigger_log,
+    )
+    assert result is None
+    assert len(trigger_log.evaluations) == 0
+
+
+def test_inform_trigger(ensemble_trigger_config: EnsembleTriggerConfig) -> None:
+    trigger = EnsembleTrigger(ensemble_trigger_config)
```

Comment:
I have a hard time parsing this test, but basically I am trying to understand whether this also tests the state across inform calls. Do we have a test case that checks whether storing the old decisions (that some trigger is `True` already when starting inform) works as expected? If not, can you add one? Alternatively, you could also extend the `test_find_next_trigger_index` above maybe with a dict that already contains true

Reply:
For the first part, yes `test_inform_multi_batch` should test exactly that. For the latter, yes, it makes sense to explicitly tests that in `find_next_trigger_index`